modules/apps/push-notifications/push-notifications-test/src/testIntegration/java/com/liferay/push/notifications/sender/test/FirebasePushNotificationsSenderTest.java

```diff
@@ -0,0 +1,794 @@
+			JSONUtil.put(
+				"operation", "create"
+			).put(
+				"registration_ids", destinationTokens
+			).toString(),
+			jsonObject.toString(), JSONCompareMode.STRICT_ORDER);
+
+		_clientAndServer.verify(
+			HttpRequest.request(
+			).withBody(
+				jsonObject.toString()
+			).withHeader(
+				"access_token_auth", "true"
+			).withHeader(
+				"project_id", _PROJECT_NUMBER
+			).withHeader(
+				"Authorization", "Bearer " + accessToken
+			).withHeader(
+				"Content-Type", "application/json"
+			).withMethod(
+				"POST"
+			).withPath(
+				"/fcm/notification"
+			),
+			VerificationTimes.once());
+
+		return jsonObject.getString("notification_key_name");
+	}
+
+	private void _verifyGroupRequestInteractions(
+		String accessToken, boolean createGroupInteraction,
+		boolean removeGroupInteraction) {
+
+		int times = 0;
+
+		if (createGroupInteraction) {
+			times++;
+		}
+
+		if (removeGroupInteraction) {
+			times++;
+		}
+
+		_clientAndServer.verify(
+			HttpRequest.request(
+			).withHeader(
+				"access_token_auth", "true"
+			).withHeader(
+				"project_id", _PROJECT_NUMBER
+			).withHeader(
+				"Authorization", "Bearer " + accessToken
+			).withHeader(
+				"Content-Type", "application/json"
+			).withMethod(
+				"POST"
+			).withPath(
+				"/fcm/notification"
+			),
+			VerificationTimes.exactly(times));
+	}
+
+	private void _verifyRemoveGroupRequest(
+		String accessToken, List<String> destinationTokens, String groupId,
+		String groupName) {
+
+		_clientAndServer.verify(
+			HttpRequest.request(
+			).withBody(
+				JSONUtil.put(
+					"notification_key", groupId
+				).put(
+					"notification_key_name", groupName
+				).put(
+					"operation", "remove"
+				).put(
+					"registration_ids", destinationTokens
+				).toString()
+			).withHeader(
+				"access_token_auth", "true"
+			).withHeader(
+				"project_id", _PROJECT_NUMBER
+			).withHeader(
+				"Authorization", "Bearer " + accessToken
+			).withHeader(
+				"Content-type", "application/json"
+			).withMethod(
+				"POST"
+			).withPath(
+				"/fcm/notification"
+			),
+			VerificationTimes.once());
+	}
+
+	private void _verifySendNotificationRequest(
+		String accessToken, JSONObject notificationJSONObject) {
+
+		_clientAndServer.verify(
+			HttpRequest.request(
+			).withBody(
+				notificationJSONObject.toString()
+			).withHeader(
+				"Authorization", "Bearer " + accessToken
+			).withHeader(
+				"Content-Type", "application/json"
+			).withMethod(
+				"POST"
+			).withPath(
+				"/v1/projects/" + _PROJECT_ID + "/messages:send"
+			),
+			VerificationTimes.once());
+	}
+
+	private static final String _PID =
+		"com.liferay.push.notifications.sender.firebase.internal." +
+			"configuration.FirebasePushNotificationsSenderConfiguration";
+
+	private static final String _PROJECT_ID = RandomTestUtil.randomString();
+
+	private static final String _PROJECT_NUMBER = RandomTestUtil.randomString();
+
+	private ClientAndServer _clientAndServer;
+
+	@Inject(
```

Comment:
Have you considered using `PushNotificationsDeviceLocalService` instead? That is the documented API to interact with push notifications. I think it would be better to use it as subject under test.